code/camera/cameradisplay.cpp

```diff
@@ -117,6 +127,28 @@ void CameraDisplay::update_zoom(int value) {
     Q_EMIT zoom_changed(zoom_factor);
 }
 
+void CameraDisplay::gridSelectChanged(int weight_index) {
+        weightSelected = m_ui->weight_list->currentText();
+     //   m_ui->grid_display->selectRobotPosition(weightSelected);
+    Q_EMIT select_position(weightSelected);
+    #ifndef NDEBUG
+        qDebug() << "Grid selector option changed to " << QString(weightSelected);
```

Comment:
`weightSelected` should be named `m_selected_weight` or something, and I don't think you need `QString` around it to use with `qDebug()`